### PR TITLE
CLN: Pandas deprecations rebased

### PR DESCRIPTION
**Pull request by bashtage**
Rebases #1542 

closes #1542

### CODE REVIEW

File: statsmodels/tsa/base/tsa_model.py [hunk 1]
```diff
@@ -120,7 +117,8 @@ def _get_predict_start(self, start):
         Start can be a string or an integer if self.data.dates is None.
         """
         dates = self.data.dates
-        if isinstance(start, str):
+        if not isinstance(start, (int, long)):
+            start = str(start)
```
Comment (bashtage) [inline statsmodels/tsa/base/tsa_model.py]: This is one fix I needed to make -- modern Pandas will output dates in various forms, e.g. a pandas `Period`

File: statsmodels/tsa/base/tsa_model.py [hunk 2]
```diff
@@ -49,6 +51,8 @@ def _init_dates(self, dates, freq):
             if (not datetools._is_datetime_index(dates) and
                     isinstance(self.data, data.PandasData)):
                 try:
+                    if is_numeric_dtype(dates):
+                        raise ValueError
```
Comment (bashtage) [inline statsmodels/tsa/base/tsa_model.py]: This change was needed to ensure that indices like `[1,2,3...]` are not converted to dates, which `to_datetime` is happy to convert to (nano?) milliseconds past 1-1-1970.

**Comment by josef-pkt** [comment]
@bashtage  Did you see anything that needs to go into 0.8?

I prefer to merge this for 0.9, in case there are subtle changes that won't be covered by unit tests, and might be too dangerous between rc and final release.

Looks good overall, but there are many parts where I'm not familiar with the code or details, so I don't even have a very good guess at the impact or side effects.

**Comment by bashtage** [comment]
It isn't that important -- rebased to it can get in, but doesn't but nothin broken in 0.8 without.

**Comment by bashtage** [comment]
@ChadFulton I think this solves the same issue.

xref #3113

**Comment by ChadFulton** [comment]
Great! I'd forgotten about this PR. Looks like it will help out a lot generally, and I think it solves the same important issue of people using integer Pandas indexes. If it's not going to be merged for 0.8, then we should merge #3113 which at least prevents errors popping up in a weird place (forecasting even though the model can be created and fit just fine).

Also, this PR introduces a backwards incompatible change, in that `endog = pd.Series([...])` works in the current code but would raise a `ValueError` with this code. On the one hand, it wasn't supposed to work, but on the other hand - was that ever even documented anywhere?

**Comment by josef-pkt** [comment]
I will look at this again for 0.8. I'm starting to think this better goes into 0.8.

**Comment by josef-pkt** [comment]
After reading through all of it, I still don't understand this part of the code. 

I'm merging in the hope that everything is fine, and assume we have enough test coverage for this.

@bashtage Thanks for updating the PR.

**Pull request merged**